src/test/java/com/redhat/darcy/webdriver/TakeScreenshotTest.java

```diff
@@ -1,49 +1,97 @@
+import org.mockito.InOrder;
 import org.openqa.selenium.NoSuchWindowException;
 import org.openqa.selenium.OutputType;
 
-@RunWith(MockitoJUnitRunner.class)
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStream;
+
+@RunWith(JUnit4.class)
 public class TakeScreenshotTest {
-    @Mock
-    TargetedWebDriver mockedDriver;
-
-    Browser browser;
-
-    @Before
-    public void setupBrowser() {
-        if (browser == null) {
-            browser = new WebDriverBrowser(mockedDriver,
-                    new StubWebDriverParentContext(),
-                    new StubWebDriverElementContext());
-        }
+    @Test
+    public void shouldTakeScreenshotAndWriteToOutputStream() throws IOException {
+        TargetedWebDriver mockedDriver = mock(TargetedWebDriver.class);
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        Browser browser = new WebDriverBrowser(mockedDriver,
+                new StubWebDriverParentContext(),
+                new StubWebDriverElementContext());
+
+        byte[] data = new byte[] { 1, 2, 3 };
+
+        when(mockedDriver.getScreenshotAs(OutputType.BYTES))
+                .thenReturn(data);
+
+        browser.takeScreenshot(baos);
+
+        assertThat(baos.toByteArray(), equalTo(data));
     }
 
     @Test
-    public void shouldTakeScreenshot() {
-        browser.takeScreenshot();
-        verify(mockedDriver).getScreenshotAs(OutputType.FILE);
+    public void shouldTakeScreenshotAndWriteToOutputStreamThenFlushAndCloseInOrder() throws IOException {
+        TargetedWebDriver mockedDriver = mock(TargetedWebDriver.class);
+        OutputStream outputStream = mock(OutputStream.class);
+        Browser browser = new WebDriverBrowser(mockedDriver,
+                new StubWebDriverParentContext(),
+                new StubWebDriverElementContext());
+
+        InOrder inOrder = inOrder(outputStream);
+
+        browser.takeScreenshot(outputStream);
+
+        inOrder.verify(outputStream).write(any(byte[].class));
+        inOrder.verify(outputStream).flush();
+        inOrder.verify(outputStream).close();
     }
 
     @SuppressWarnings("unchecked")
     @Test(expected = FindableNotPresentException.class)
-    public void shouldThrowFindableNotPresentExceptionIfDriverIsNotPresent() {
-        when(mockedDriver.getScreenshotAs(OutputType.FILE))
+    public void shouldThrowFindableNotPresentExceptionIfDriverIsNotPresent() throws IOException {
+        TargetedWebDriver driver = mock(TargetedWebDriver.class);
+        OutputStream outputStream = mock(OutputStream.class);
+        Browser browser = new WebDriverBrowser(driver,
+                new StubWebDriverParentContext(),
+                new StubWebDriverElementContext());
+
+        when(driver.getScreenshotAs(OutputType.BYTES))
                 .thenThrow(NoSuchWindowException.class);
-        browser.takeScreenshot();
+        browser.takeScreenshot(outputStream);
+
+        verify(outputStream).close();
```

Comment:
Ah, yeah. Derp. Also, the first exception test shouldn't even need to check the `OutputStream` methods because the exception gets thrown due to not being able to find a driver and doesn't even get to any `OutputStream` operations.